modules/substrate/src/lib.rs

```diff
@@ -222,18 +236,19 @@ impl<T: Trait> Module<T> {
 		}
 	}
 
-	/// Return the latest header which enacts an authority set change
-	/// and still needs a finality proof.
+	/// Returns a list of headers which require finality proofs.
 	///
-	/// Will return None if there are no headers which are missing finality proofs.
-	pub fn requires_justification() -> Option<BridgedHeader<T>> {
+	/// These headers require proofs because they enact authority set changes.
+	pub fn requires_justification() -> Vec<BridgedHeader<T>> {
 		let storage = PalletStorage::<T>::new();
-		let hash = storage.unfinalized_header()?;
-		let imported_header = storage.header_by_hash(hash).expect(
-			"We write a header to storage before marking it as unfinalized, therefore
-			this must always exist if we got an unfinalized header hash.",
-		);
-		Some(imported_header.header)
+		let hashes = storage.unfinalized_headers();
```

Comment:
Probably use some other name? There may be a ton of unfinalized headers but only few of these would enact new authorities set => would require implicit justifications. Looking at the code it seems that the function is returning all `unfinalized_headers` here :)